src/nations/nations.cpp

```diff
@@ -906,15 +936,29 @@ std::vector<dcon::political_party_id> get_active_political_parties(sys::state& s
 	return parties;
 }
 
-void monthly_adjust_relationship(sys::state& state, dcon::nation_id a, dcon::nation_id b, float delta) {
+dcon::diplomatic_relation_id get_diplomatic_relation(sys::state& state, dcon::nation_id a, dcon::nation_id b, bool force_create) {
```

Comment:
The way that dcon works is that it is safe to look up values from the invalid handle, and such values will always be zero. Thus, when there was no relationship object between nations, looking up the relationship would return the invalid handle, and then looking up the relationship value from that would return 0. This helped cut down on the number of relationship objects. If we created a relationship object every time we checked the relations between two nations we would quickly head towards n squared such objects.

Reply:
I do see that you have force create as an extra parameter, so I realize that this won't actually be creating such relationships everywhere. I just makes my danger sense tingle. So my comment above isn't a blocking objection.